MIGRATION.md

```diff
@@ -41,6 +41,8 @@ API changes in MoveIt releases
 - Moved the class `MoveItErrorCode` from both `moveit_ros_planning` and `moveit_ros_planning_interface` to `moveit_core`. The class now is in namespace `moveit::core`, access via `moveit::planning_interface` or `moveit_cpp::PlanningComponent` is deprecated.
 - End-effector markers in rviz are shown only if the eef's parent group is active _and_ the parent link is part of that group. Before, these conditions were _OR_-connected.
   You might need to define additional end-effectors.
+- Removed `ConstraintSampler::project()` as there was no real difference to `sample()`.
```

Comment:
@mamoll did we actually have the discussion around this and what the correct path moving forward was?

Reply:
There was a discussion on [the MoveIt 1 PR](https://github.com/ros-planning/moveit/pull/3170). The current API reflects more clearly what is current practice. Ideally, the implementations would have been fixed to match the intended semantics of the API, but that seemed too difficult. So instead the API has been changed to match the implementation.

Reply:
I also think that it makes sense to merge this, and that at least the technical discussion has been done. There is no obvious reason to me why we should reject this change for MoveIt 2.